### ISSUE TITLE
resizing behaves wrong when  some  props(maxWidth, maxHeight, minWidth,  minHeight) has a value

### ISSUE THREAD

**Issue by flwzz** [bokuweb/react-rnd#790]
yes
 not a browser bug

### Overview of the problem

I'm using react-rnd **version** [10.3.0]   
     
My **browser** is:    Chrome  version=90.0.4430.212 x64 on windows10
 
I am sure this issue is **not a duplicate**?   yes

### Reproduced project

https://codesandbox.io/s/upbeat-bohr-41o6x

### Description
when you drag the left side to resizing, the box's transform-origin is on the top-left. when resizing is end,  the box  jump to the previous position on the right.  Everything is ok when I tried it in version 10.2.4.

### Steps to Reproduce

1. First Step
2. Second Step
3. and so on...

**Comment by bokuweb** [comment]
@flwzz Thanks. I guess it has been fixed by v10.3.1. please try it.

**Comment by slavagoreev** [comment]
Draggable doesn't update state if you change it implicitly. You may do this with some kind of this:
```
const Draggable: Component = this.rndRef.current?.draggable;
Draggable?.setState({ x: layoutProps.x, y: layoutProps.y });
```
